### DISCUSSION TITLE
Communication between Client and ModelServer

### DISCUSSION THREAD

**Discussion by e-lynch-theia** [eclipse-emfcloud/emfcloud#225] [ModelServer]
Hello,
I have created a very simple modelServer in eclipse, and a typescript frontend which works with an emf model.  I have created a command on the frontend which asks the modelServer to do something.  The result does not update the emf model due to a particular reason which I want to tell the UI.  How can I return this result - do I need to have an EMF Result object and populate this object and thus result in a model change? - or is there another way?
Thanks
Evelyn

**Comment by eneufeld** [comment]
Hi @e-lynch-theia ,

so you are calling e.g edit on the modelserver. If this call fails, you should get a return value from the call which you then need to evaluate and forward to the ui.

Could you maybe provide more details and explain what you are calling and expecting to return in your scenario?

Best,
Eugen
